### ISSUE TITLE
Abstract class check fails for multiple inheritance

### ISSUE THREAD

**Issue by erictraut** [microsoft/pyright#470]
This bug was posted on Gitter by @yarbu.

```
import abc

class InterfaceA(abc.ABC):
  @abc.abstractmethod
  def a(self) -> None: print('InterfaceA.a')

class MixinA(InterfaceA):
  def a(self) -> None: print('MixinA.a')

class InterfaceAB(InterfaceA):
  @abc.abstractmethod
  def b(self) -> None: print('InterfaceAB.b')

class ClassAB(InterfaceAB, MixinA):
  def b(self) -> None: print('ClassAB.b')

ab = ClassAB()
ab.a()
```

**Comment by erictraut** [comment]
Thanks for reporting this bug. It will be fixed in the next release of Pyright.

**Comment by erictraut** [comment]
This is fixed in 1.1.18, which I just published.

**Comment by yarbu** [comment]
Please consider reopening this bug. Pyright reports 'Cannot instantiate abstract class "ClassAB" "InterfaceA.a" is abstract' for versions 1.1.25 to 1.1.30.

**Comment by erictraut** [comment]
Thanks for reminding me. I had made a note to myself to fix this once I added true MRO support for method lookup, but I forgot to do it. 

I've fixed the issue, and it will be in the next published version of pyright.

**Comment by erictraut** [comment]
This is implemented in 1.1.31, which I just published.
